pkg/netstate/netstate.go

```
@@ -99,6 +134,13 @@ func (s *Server) List(ctx context.Context, req *pb.ListRequest) (*pb.ListRespons
 func (s *Server) Delete(ctx context.Context, req *pb.DeleteRequest) (*pb.DeleteResponse, error) {
 	s.logger.Debug("entering netstate delete")
 
+	xAPIKeyBytes := []byte(req.XApiKey)
+	if err := validateAuth(xAPIKeyBytes); err != nil {
+		s.logger.Error("unauthorized request")
+
+		return &pb.DeleteResponse{}, nil
```

should you return the error ?